### PR TITLE
add InsertContainmentTreeEntryForSequenceId

### PR DESCRIPTION
**Pull request by antipovi-draeger**
Add manipulation InsertContainmentTreeEntryForSequenceId

# Checklist

The following aspects have been respected by the author of this pull request, confirmed by both pull request assignee **and** reviewer:

* Changelog update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [ ] Reviewer
* README update (necessity checked and entry added or not added respectively)
  * [x] Pull Request Assignee
  * [ ] Reviewer

### CODE REVIEW

**Review by ldeichmann — CHANGES_REQUESTED** [review]
Prefer the BPKP definition

File: src/t2iapi/device/service.proto [hunk 1]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+    CONTAINMENT TREE ENTRY is already present in the current MDIB, only its handle shall be returned, no further
+    action is required.
+
+    Hint: Devices which use static handles for descriptors do not have to lookup the previous handles since
+    they stay the same.
+
+    Example 1: returning the same handle for the Metric
+    - device presented an MDIB with the Metric handle "metric_1" in @SequenceID "123"
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - metric with the handle "metric_1" is still present in the current MDIB
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:199]:
```suggestion
    - metric with the handle "metric_1" is still present in the current MDIB and represents the *same* metric as in @SequenceID "123"
```

File: src/t2iapi/device/service.proto [hunk 2]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
      */
     rpc IndicateTimeOfNextCalibrationToUser (BasicHandleRequest)
         returns (BasicResponse);
+
+    /*
+    Make available the CONTAINMENT TREE ENTRY(as defined in IEEE Std 11073-10207-2017) that was previously
+    represented by the provided handle in the MDIB with the provided @SequenceId. If the CONTAINMENT TREE ENTRY is
+    absent in the current MDIB, it shall be inserted and its (new) handle shall be returned. If the
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:189]:
```suggestion
    absent in the current MDIB, it shall be inserted and its handle in the current MDIB shall be returned. If the
```

File: src/t2iapi/device/service.proto [hunk 3]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - metric with the handle "metric_1" is still present in the current MDIB
+    - InsertContainmentTreeEntryForSequenceId with handle="metric_1" and sequence_id="123" is executed
+    - "metric_1" is returned since the descriptor with this handle it is still available in the current MDIB
+    - no further action from the device is required
+
+    Example 2: returning a new handle for the MDS:
+    - device presented an MDIB with MDS handle "mds0" in @SequenceID "123"
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - MDS with the handle "mds0" is still present in the current MDIB but it's handle changed to "mds10"
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:207]:
```suggestion
    - MDS which had the Handle "mds0" in @SequenceID "123" is still present in the current MDIB, but it's handle changed to "mds10"
```

File: src/t2iapi/device/service.proto [hunk 4]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+    - InsertContainmentTreeEntryForSequenceId with handle="mds0" and sequence_id="123" is executed
+    - "mds10" is returned since the descriptor handle for that MDS has changed
+    - no further action from the device is required
+
+    Example 3: inserting the Channel into the current MDIB and returning the (new) handle:
+    - device presented the Channel with handle "channel123" in @SequenceID "123"
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - channel with handle "channel123" is no longer present in the current MDIB
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:215]:
```suggestion
    - Channel which had the Handle "channel123" in @SequenceID "123" is no longer present in the current MDIB
```

File: src/t2iapi/device/service.proto [hunk 5]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+    - InsertContainmentTreeEntryForSequenceId with handle="channel123" and sequence_id="123" is executed
+    - channel is inserted into the device's current MDIB
+    - the (new) handle of the inserted channel is returned
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:218]: Maybe be concrete about the handle here as well
```suggestion
    - Channel is inserted into the device's current MDIB with handle "channel789"
    - the handle "channel789" is returned
```

File: src/t2iapi/device/service.proto [hunk 6]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+    - InsertContainmentTreeEntryForSequenceId with handle="metric_1" and sequence_id="123" is executed
+    - "metric_1" is returned since the descriptor with this handle it is still available in the current MDIB
+    - no further action from the device is required
+
+    Example 2: returning a new handle for the MDS:
+    - device presented an MDIB with MDS handle "mds0" in @SequenceID "123"
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - MDS with the handle "mds0" is still present in the current MDIB but it's handle changed to "mds10"
+    - InsertContainmentTreeEntryForSequenceId with handle="mds0" and sequence_id="123" is executed
+    - "mds10" is returned since the descriptor handle for that MDS has changed
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:209]:
```suggestion
    - the handle "mds10" is returned as the descriptor handle for that MDS has changed
```

File: src/t2iapi/device/service.proto [hunk 7]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+
+    Hint: Devices which use static handles for descriptors do not have to lookup the previous handles since
+    they stay the same.
+
+    Example 1: returning the same handle for the Metric
+    - device presented an MDIB with the Metric handle "metric_1" in @SequenceID "123"
+    - device changes @SequenceID to "789" and presents a new MDIB
+    - metric with the handle "metric_1" is still present in the current MDIB
+    - InsertContainmentTreeEntryForSequenceId with handle="metric_1" and sequence_id="123" is executed
+    - "metric_1" is returned since the descriptor with this handle it is still available in the current MDIB
```
Comment (ldeichmann) [inline src/t2iapi/device/service.proto:201]:
```suggestion
    - the handle "metric_1" is returned as the descriptor with this handle it is still available in the current MDIB
```

**Review by ldeichmann — CHANGES_REQUESTED** [review]

File: src/t2iapi/device/service.proto [hunk 8]
```diff
@@ -182,4 +182,41 @@ service DeviceService {
...
+
+    /*
+    Make available the CONTAINMENT TREE ENTRY(as defined in IEEE Std 11073-10207-2017) that was previously
+    represented by the provided handle in the MDIB with the provided @SequenceId. If the CONTAINMENT TREE ENTRY is
+    absent in the current MDIB, it shall be inserted and its (new) handle shall be returned. If the
+    CONTAINMENT TREE ENTRY is already present in the current MDIB, only its handle shall be returned, no further
+    action is required.
+
+    Hint: Devices which use static handles for descriptors do not have to lookup the previous handles since
+    they stay the same.
```
[the remark this answers is not in the document]
Comment (antipovi-draeger) [reply src/t2iapi/device/service.proto:194]: i like it

**Review by pstaeck — COMMENTED** [review]
I agree with Lukas remarks. No further findings from my side.

File: src/t2iapi/device/device_responses.proto [hunk 9]
```diff
@@ -63,3 +63,13 @@ message GetComponentHwVersionResponse {
     BasicResponse status = 1;    // status of the rpc
     string hardware_version = 2; // Hardware version of the requested device component
 }
+
+/*
+Response containing the descriptor handle which is representing the requested
+BICEPS CONTAINMENT TREE ENTRY (as defined in IEEE Std 11073-10700-2022 in the device's current MDIB.
```
Comment (pstaeck) [inline src/t2iapi/device/device_responses.proto:69]: a braket is missing

**Review by pstaeck — CHANGES_REQUESTED** [review]

File: src/t2iapi/device/device_responses.proto (hunk 9 above)
Comment (antipovi-draeger) [reply src/t2iapi/device/device_responses.proto]: done

**Review by pstaeck — APPROVED** [review]

**Review by ldeichmann — APPROVED** [review]

**Pull request merged**